packages/jsii-pacmak/lib/targets/golang/documentation.ts

```diff
@@ -1,15 +1,73 @@
 import { Docs } from 'jsii-reflect';
-import { EmitContext } from './emit-context';
+import { Rosetta } from 'jsii-rosetta';
+import { CodeMaker } from 'codemaker';
 
 export class Documentation {
-  public constructor(private readonly docs: Docs) {}
+  public constructor(
+    private readonly code: CodeMaker,
+    private readonly rosetta: Rosetta,
+  ) {}
 
-  public emit({ code }: EmitContext): void {
-    // TODO: Translate code examples to Golang with Rosetta (not implemented there yet)
-    code.line(`// ${this.docs.summary}`);
-    code.line('//');
-    for (const line of this.docs.remarks.split('\n')) {
-      code.line(`// ${line}`);
+  /**
+   * Emits all documentation depending on what is available in the jsii model
+   *
+   * Used by all kind of members + classes, interfaces, enums
+   * Order should be
+   * Summary + Remarks
+   * Returns
+   * Examples TODO rosetta stuff
+   * Link
+   * Stability/Deprecation description
+   */
+  public emit(docs: Docs): void {
+    console.log(this.rosetta); // Placeholder, until rosetta is used
+
+    if (docs.toString() !== '') {
+      const lines = docs.toString().split('\n');
+      for (const line of lines) {
+        this.code.line(`// ${line}`);
+      }
+    }
+
+    if (docs.returns !== '') {
+      this.code.line(`//`);
+      this.code.line(`// Returns: ${docs.returns}`);
+    }
+
+    if (docs.example !== '') {
+      this.code.line(`//`);
+      // TODO: Translate code examples to Golang with Rosetta (not implemented there yet)
+      this.code.line('// TODO: EXAMPLE');
+      this.code.line(`//`);
+    }
+
+    if (docs.link !== '') {
+      this.code.line(`// See: ${docs.link}`);
+      this.code.line(`//`);
+    }
+
+    this.emitStability(docs);
+  }
+
+  public emitStability(docs: Docs): void {
+    const stability = docs.stability;
+    if (stability && docs.shouldMentionStability()) {
+      if (docs.deprecated) {
+        this.code.line(`// Deprecated: ${docs.deprecationReason}`);
+      } else {
+        this.code.line(`// ${this.code.toPascalCase(stability)}.`);
+      }
     }
   }
 }
+
+// "docs": {
```

Comment:
Is keeping this example around intentional? Is this structure declared on the jsii-reflect.Docs type?

Reply:
was meant as reference. As discussed offline, the jsii-reflect lib doesn't seem to have a complete API for the underlying doc object.